src/librustc_data_structures/bit_set.rs

```diff
@@ -0,0 +1,1029 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use array_vec::ArrayVec;
+use indexed_vec::{Idx, IndexVec};
+use rustc_serialize;
+use std::fmt;
+use std::iter;
+use std::marker::PhantomData;
+use std::mem;
+use std::slice;
+
+pub type Word = u64;
+pub const WORD_BYTES: usize = mem::size_of::<Word>();
+pub const WORD_BITS: usize = WORD_BYTES * 8;
+
+/// A fixed-size bitset type with a dense representation.
+///
```

Comment:
The documentation for `struct IdxSet` attempted to explain the relationship between `T` and the `IdxSet<T>`. The doc-comment here does not do that. Would you mind skimming over the doc-comments in the now deleted `indexed_set.rs` and porting over any, like that for `struct IdxSet`, that add useful info for a newcomer trying to understand the type?

Reply:
This is the old comment:
```
/// Represents a set of some element type E, where each E is identified by some
/// unique index type `T`.
///
/// In other words, `T` is the type used to index into the bitvector
/// this type uses to represent the set of object it holds.
```
I always found the part about `E` confusing, which is why I dropped the comment. And given that the new type name is `BitSet` it should be clear that the set elements are bits. But I'm happy to add something about `T` being an index type; I can do that for all the data structures in this file.